src/app/page/Recipient/RequestsList.tsx

```diff
@@ -0,0 +1,235 @@
+    width: "1.5rem",
+  },
+  cardHeader: {
+    alignItems: "end",
+  },
+  cardActions: {
+    justifyContent: "center",
+  },
+  cardBottomToggle: {
+    padding: ".5rem",
+    "& p": {
+      fontWeight: "600",
+      fontSize: ".8rem",
+    },
+  },
+  fallback: {
+    marginTop: "2rem",
+  },
+  confirmationImage: {
+    width: "calc(100% - 4rem)",
+    margin: "1rem 2rem",
+  },
+  volunteerName: {
+    display: "flex",
+    margin: "auto",
+    width: "100%",
+    justifyContent: "center",
+    alignItems: "center",
+    "& .MuiAvatar-root": {
+      margin: "0 .5rem",
+    },
+  },
+}));
+
+type Props = {
+  missions: MissionInterface[];
+  fallback: string;
+};
+
+export default function ({ fallback, missions }: Props) {
+  const classes = useStyles();
+
+  if (missions.length < 1) {
+    return (
+      <Typography className={classes.fallback} align="center" variant="body1" color="textSecondary">
+        {fallback}
+      </Typography>
+    );
+  }
+
+  return (
+    <>
+      {missions.map((mission) => (
+        <RecipientMissionCard key={mission.uid} mission={mission} />
+      ))}
+    </>
+  );
+}
+
+function RecipientMissionCard({ mission }: { mission: MissionInterface }) {
+  const classes = useStyles();
+  const [open, setOpen] = useState(false);
+  const toggle = () => setOpen(!open);
+
+  return (
+    <Card className={classes.card}>
+      <TopBar
+        status={mission.status}
+        fundedStatus={mission.fundedStatus}
+        createdDate={mission.createdDate}
+      ></TopBar>
+      <CardActionArea onClick={toggle}>
+        <CardHeader
+          className={classes.cardHeader}
+          avatar={<AppleIcon className={classes.appleIcon} />}
+          title={<Typography variant="h4">FOOD BOX DELIVERY</Typography>}
+          subheader={mission.details?.map((r) => (
+            <Typography key={r.resourceUid} variant="subtitle2">
+              {r.quantity} x {r.displayName}
+            </Typography>
+          ))}
+        />
+      </CardActionArea>
+      <Collapse in={open}>
+        <Details mission={mission} />
+        {mission.status === Status.delivered && <ConfirmDelivery mission={mission} />}
+      </Collapse>
+      <CardActionArea onClick={toggle} className={classes.cardBottomToggle}>
+        <Typography align="center" color="primary">
+          {open ? "HIDE DETAILS" : "MORE DETAILS"}
+        </Typography>
+      </CardActionArea>
+    </Card>
+  );
+}
+
+type TopBarProps = {
+  status: Status;
+  fundedStatus: FundedStatus;
+  createdDate: string;
+  className?: string;
+};
+
+function Icon({ fundedStatus, status, ...rest }: any) {
+  return status === Status.started ? (
+    <PlayCircleFilled {...rest} />
+  ) : status === Status.delivered ? (
+    <NotificationImportant {...rest} />
+  ) : status === Status.succeeded ? (
+    <CheckCircle {...rest} />
+  ) : fundedStatus === FundedStatus.notfunded ? (
+    <HourglassFull {...rest} />
+  ) : (
+    <FiberNew {...rest} />
+  );
+}
```

Comment:
My eyes! haha jk

Can we just use a hash map(?) like this:

```
const ICON_STATUS = {
  [Status.started]: PlayCircleFilled,
  [Status.delivered]: <NotificationImportant,
  [Status.succeeded]: CheckCircle,
  [FundedStatus.notfunded]: HourglassFull
  ['default']: FiberNew
}

// and then use it something like
const StatusIcon = (status) => {
	if (!(Object.keys(ICON_STATUS).contains(status)) {
		return ICON_STATUS['default']
	}

    return ICON_STATUS[status]
}

//and then 
//...

<StatusIcon [...your props ]/>
```

Some of these ternary chains are really hard to follow

Reply:
I wanted to do a map or even just a switch but I need to account for two different status objects which causes problems... like when would fundedstatus override the regular status?

Would and `if/else` look cleaner?

```ts
if (status === STATUS...) return <Icon/>
if (status === STATUS...) return <Icon/>
if (status === STATUS...) return <Icon/>
if (status === STATUS...) return <Icon/>
if (fundedstatus === FUNDEDSTATUS...) return <Icon/>
```

Reply:
I would prefer switch due to both:
1. Switch is made for this use case, map is just a convenient way.
2. You actually create an object every time you render using the map approach, we had this discussion earlier.

Reply:
neither the switch nor map will work unless we can consolidate all statuses to single property.
Need to be able to test for both status and funded status which I can't really directly do in a switch.
Gonna try the `if` block approach